patch/field_type.go

```diff
@@ -0,0 +1,201 @@
+	// Cast Getter signature
+	case *types.Signature:
+		n, ok := parent.(*ast.FuncDecl)
+		if !ok {
+			log.Printf("Warning: unexpected type for getter: %v `%T`", obj, parent)
+			break
+		}
+		if l := len(n.Type.Results.List); l != 1 {
+			log.Printf("Warning: unexpected return count for getter: %v `%d`", obj, l)
+			return
+		}
+		if !castDecl(n.Type.Results.List[0]) {
+			log.Printf("Warning: unsupported fieldType type: %T `%s`", n.Type.Results.List[0].Type, fieldType)
+		}
+		return
+	}
+}
+
+func (p *Patcher) patchTypeUsage(id *ast.Ident, obj types.Object) {
+	desiredType, ok := p.fieldTypes[obj]
+	if !ok {
+		return
+	}
+	var originalType string
+	switch t := obj.Type().(type) {
+	case *types.Basic:
+		originalType = t.Name()
+	case *types.Signature:
+		if t.Results().Len() != 1 {
+			return
+		}
+		originalType = t.Results().At(0).Type().String()
+	}
+	cast := func(as string, expr ast.Expr) ast.Expr {
+		return &ast.CallExpr{
+			Fun: &ast.Ident{
+				Name: as,
+			},
+			Args: []ast.Expr{expr},
+		}
+	}
+
+	usageNode := p.findParentNode(id)
+	parentNode := p.findParentNode(usageNode)
+
+	switch usage := usageNode.(type) {
+	case *ast.SelectorExpr:
+		switch parentExpr := parentNode.(type) {
+		case *ast.AssignStmt:
+			if len(parentExpr.Lhs) != len(parentExpr.Rhs) {
+				return
+			}
+			for i := range parentExpr.Lhs {
+				if parentExpr.Lhs[i] == usage {
+					parentExpr.Rhs[i] = cast(desiredType, parentExpr.Rhs[i])
+					return
+				}
+			}
+			for i := range parentExpr.Rhs {
+				if parentExpr.Rhs[i] == usage {
+					parentExpr.Rhs[i] = cast(originalType, parentExpr.Rhs[i])
+					return
+				}
+			}
+		case *ast.CallExpr:
+			for i := range parentExpr.Args {
+				if parentExpr.Args[i] == usage {
+					parentExpr.Args[i] = cast(originalType, parentExpr.Args[i])
+					return
+				}
+			}
+			parent := p.findParentNode(parentExpr)
+			assign, isAssign := parent.(*ast.AssignStmt)
+			if parentExpr.Fun == usage && isAssign {
+				for i := range assign.Rhs {
+					if assign.Rhs[i] == parentExpr {
+						assign.Rhs[i] = cast(originalType, assign.Rhs[i])
+						return
+					}
+				}
+			}
+			call, isCall := parent.(*ast.CallExpr)
+			if isCall {
+				for i := range call.Args {
+					if call.Args[i] == parentExpr {
+						call.Args[i] = cast(originalType, call.Args[i])
+						return
+					}
+				}
+			}
+			for i, v := range parentExpr.Args {
+				if v == usage {
+					parentExpr.Args[i] = cast(originalType, usage)
+					return
+				}
+			}
+		case *ast.BinaryExpr:
+			if parentExpr.X == usage {
+				parentExpr.X = cast(originalType, parentExpr.X)
+			}
+			if parentExpr.Y == usage {
+				parentExpr.Y = cast(originalType, parentExpr.Y)
+			}
+		}
+	case *ast.KeyValueExpr:
+		if usage.Key == id {
+			usage.Value = cast(desiredType, usage.Value)
+			return
+		}
+		if usage.Value == id {
+			usage.Value = cast(originalType, usage.Value)
+			return
+		}
+	}
+}
+
+func packageAndName(fqn string) (pkg string, name string, isSlice bool) {
```

Comment:
Can you remove this and put it in the followup PR(s)?

Reply:
it is used to check and warn.

Reply:
Do you want the warnings to be removed ?

Reply:
Can you change it to warn if the specified type isn't a valid Go name?

Reply:
Yes, I'll do that.